code/game/turfs/closed/walls.dm

```diff
@@ -363,5 +363,17 @@
 /turf/closed/wall/metal_foam_base
 	girder_type = /obj/structure/foamedmetal
 
+/turf/closed/wall/Bumped(atom/movable/bumped_atom)
+	. = ..()
+	if(isliving(bumped_atom))
+		var/mob/living/living_bumped = bumped_atom
+		SEND_SIGNAL(living_bumped, COMSIG_LIVING_WALL_BUMP, src)
+
+/turf/closed/wall/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(isliving(gone))
+		var/mob/living/living_gone = gone
+		SEND_SIGNAL(living_gone, COMSIG_LIVING_WALL_EXITED, src)
```

Comment:
We don't need to typecast to living? The source for a signal (AKA the first argument for any SIGNAL_HANDLER proc) should work on the `/datum` level. Just have it send the signal unconditionally, DCS is already plenty inexpensive when it comes to filtering out non-listening datums.